src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -296,11 +296,14 @@ protected virtual void ValidateCreateChannelParameters(EndpointAddress remoteAdd
 
         protected override TChannel OnCreateChannel(EndpointAddress remoteAddress, Uri via)
         {
-            //EndpointAddress httpRemoteAddress = remoteAddress != null && WebSocketHelper.IsWebSocketUri(remoteAddress.Uri) ?
-            //    new EndpointAddress(WebSocketHelper.NormalizeWsSchemeWithHttpScheme(remoteAddress.Uri), remoteAddress) :
-            //    remoteAddress;
+            if (typeof (TChannel) != typeof (IRequestChannel))
+            {
+                remoteAddress = remoteAddress != null && !WebSocketHelper.IsWebSocketUri(remoteAddress.Uri) ?
+                    new EndpointAddress(WebSocketHelper.NormalizeHttpSchemeWithWsScheme(remoteAddress.Uri), remoteAddress) :
+                    remoteAddress;
+                via = !WebSocketHelper.IsWebSocketUri(via) ? WebSocketHelper.NormalizeHttpSchemeWithWsScheme(via) : via;
+            }
```

Comment:
We accept either scheme from customer code on all platforms. The difference is internally our dependencies only accept one scheme, and which one scheme that  is differs from desktop.
________________________________
From: Ron Cain<mailto:notifications@github.com>
Sent: ‎10/‎2/‎2015 5:06 AM
To: dotnet/wcf<mailto:wcf@noreply.github.com>
Cc: Matt Connew<mailto:mconnew@microsoft.com>
Subject: Re: [wcf] Fix ws vs http uri schema for WebSockets and fix timeout bug (#382)


In src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdotnet%2fwcf%2fpull%2f382%23discussion_r41014593&data=01%7c01%7cmconnew%40microsoft.com%7c0e739d26e0e34ffe341a08d2cb21d7ff%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=ShjD0RuwRU38Q3553KMoCoD2GI%2fZA%2fFl9nLNDabb3KU%3d>:

> @@ -296,11 +296,14 @@ protected virtual void ValidateCreateChannelParameters(EndpointAddress remoteAdd
>
>          protected override TChannel OnCreateChannel(EndpointAddress remoteAddress, Uri via)
>          {
> -            //EndpointAddress httpRemoteAddress = remoteAddress != null && WebSocketHelper.IsWebSocketUri(remoteAddress.Uri) ?
> -            //    new EndpointAddress(WebSocketHelper.NormalizeWsSchemeWithHttpScheme(remoteAddress.Uri), remoteAddress) :
> -            //    remoteAddress;
> +            if (typeof (TChannel) != typeof (IRequestChannel))
> +            {
> +                remoteAddress = remoteAddress != null && !WebSocketHelper.IsWebSocketUri(remoteAddress.Uri) ?
> +                    new EndpointAddress(WebSocketHelper.NormalizeHttpSchemeWithWsScheme(remoteAddress.Uri), remoteAddress) :
> +                    remoteAddress;
> +                via = !WebSocketHelper.IsWebSocketUri(via) ? WebSocketHelper.NormalizeHttpSchemeWithWsScheme(via) : via;
> +            }


Does this difference have any affect on existing Windows Store apps (e.g. in the URL's they use)? Or it is strictly an internal implementation difference (existing) customer apps would never see?

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdotnet%2fwcf%2fpull%2f382%2ffiles%23r41014593&data=01%7c01%7cmconnew%40microsoft.com%7c0e739d26e0e34ffe341a08d2cb21d7ff%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=Tnidn847blItDYL7XYs0q1UET9kbPQkWs4KNnC3An7Q%3d>.